stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+                    }
+                }
+
+                const char _Ch       = _Ctype_fac.narrow(*_First++);
+                const bool _Is_digit = ('0' <= _Ch && _Ch <= '9');
+                if (_Is_modified && _Count == 1) {
+                    if (_Ch == ':') {
+                        continue;
+                    } else if (_Is_digit && _Optional_digits > 0) {
+                        _Tz_hours = 10 * _Tz_hours + (_Ch - '0');
+                        --_Optional_digits;
+                        --_Count;
+                    } else {
+                        if (_Allow_match_fail) {
+                            break;
+                        } else {
+                            return ios_base::failbit;
+                        }
+                    }
+                } else if (_Is_digit) {
+                    auto& _Part = _Count < 2 ? _Tz_hours : _Tz_minutes;
+                    _Part       = 10 * _Part + (_Ch - '0');
+                } else {
+                    if (_Allow_match_fail) {
+                        break;
+                    } else {
+                        return ios_base::failbit;
+                    }
+                }
+            }
+
+            _Offset = 60 * _Tz_hours + _Tz_minutes;
+            if (_Negative) {
+                _Offset = -_Offset;
+            }
+            return ios_base::goodbit;
+        }
+
+        template <class _InIt>
+        _NODISCARD ios_base::iostate _Get_tz_name(
+            _InIt& _First, const ctype<typename _InIt::value_type>& _Ctype_fac, string& _Tz_name) {
+            constexpr _InIt _Last{};
+            _Tz_name.clear();
+            while (_First != _Last) {
+                const char _Ch{_Ctype_fac.narrow(*_First)};
+                if (_STD isalnum(static_cast<unsigned char>(_Ch)) || _Ch == '_' || _Ch == '/' || _Ch == '-'
+                    || _Ch == '+') {
+                    _Tz_name.push_back(_Ch);
+                    ++_First;
+                } else {
+                    break;
+                }
+            }
+            return _First == _Last ? ios_base::eofbit : ios_base::goodbit;
+        }
+
+        template <class _CharT, class _Traits, class _Alloc = allocator<_CharT>>
+        _Time_parse_fields(basic_istream<_CharT, _Traits>& _Istr, const _CharT* _FmtFirst,
+            basic_string<_CharT, _Traits, _Alloc>* _Abbrev = nullptr, minutes* _Offset = nullptr,
+            unsigned int _Subsecond_precision = 0) {
+
+            using _Myis = basic_istream<_CharT, _Traits>;
+
+            const auto& _Ctype_fac       = _STD use_facet<ctype<_CharT>>(_Istr.getloc());
+            ios_base::iostate _State     = ios_base::goodbit;
+            const _CharT* const _FmtLast = _FmtFirst + _Traits::length(_FmtFirst);
+            const typename _Myis::sentry _Ok{_Istr, true};
+
+            istreambuf_iterator _First{_Istr};
+            constexpr decltype(_First) _Last{};
+
+            if (_Ok) {
+                _TRY_IO_BEGIN

```

Comment:
I feel like this branch might deserve its own function? Feel free to ignore

Reply:
This function is massive, but that's common for iostreams - e.g. see: https://github.com/microsoft/STL/blob/62137922ab168f8e23ec1a95c946821e24bde230/stl/inc/iomanip#L290-L342

I believe we should keep this as-is - splitting this into smaller functions (and needing extra code to communicate state between them) wouldn't improve clarity. Unless there's an opportunity to extract a significant amount of code as a function with a very small interface that can be understood in isolation more easily.